### PR TITLE
Change project structure to use AdvantageKit, add simulation support to drivetrain

### PR DESCRIPTION
**Pull request by mimizh2418**
- Subsystems now use the AdvantageKit logging and subsystem IO framework
- Drivetrain simulation works, can simulate auto sequences and controller inputs
- Fixes a few logic errors discovered in simulation

### CODE REVIEW

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (DaviTheDinosaur) [inline src/main/java/org/team1540/bunnybotTank2023/Constants.java:25]: Spelled Robonauts incorrectly, go back to 2nd grade english and then rewrite comment

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (DaviTheDinosaur) [inline src/main/java/org/team1540/bunnybotTank2023/Constants.java]: Spelled Cheesy Poofs incorrectly, do the same as for the MASS constant

**Review by DaviTheDinosaur — COMMENTED** [review]
Looks good though perhaps have Zach review further since I don't have a full udnerstanding yet

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/Constants.java:25]: :skull: he doesn't get the meme https://www.chiefdelphi.com/t/team-254-presents-chezy-champs/137065

File: .gitignore
Comment (rutmanz) [inline .gitignore:4]: why are you removing the IntelliJ banner

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/Constants.java]: if this is final, when does it get changed to `Mode.REAL`?

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/Constants.java:26]: where are these actually used?

File: src/main/java/org/team1540/bunnybotTank2023/Robot.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/Robot.java]: why only when on real field?

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/Drivetrain.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/Drivetrain.java:59]: Is this the way PercentOutput converts to voltage or do they do some kind of calculation on the actual input voltage?

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java:23]: Why are you making these negative instead of inverting the motors? Presumably if your controller inputs need to be flipped, so would autos

File: src/main/java/org/team1540/bunnybotTank2023/io/drivetrain/DrivetrainIOSim.java
Comment (rutmanz) [inline src/main/java/org/team1540/bunnybotTank2023/io/drivetrain/DrivetrainIOSim.java:33]: Maybe make this actually do time calculations

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/Constants.java:26]: drivetrain simulation

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/Constants.java]: It's changed manually (yeah it's kinda scuffed)

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/Drivetrain.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/Drivetrain.java:59]: Voltage compensation is enabled, which allows this to have the same behavior as PercentOutput

File: src/main/java/org/team1540/bunnybotTank2023/Constants.java
Comment (rutmanz) [reply src/main/java/org/team1540/bunnybotTank2023/Constants.java]: I'd say that's more than just "kinda scuffed," I won't approve this without that fixed. You have a way to tell if the robot is real or not, this variable should only be for choosing between sim and replay.

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java:23]: These are simulation changes, I was messing around with teleop controls and noticed that they were inverted. Autos work fine, so maybe the controller is inverted?

File: src/main/java/org/team1540/bunnybotTank2023/Robot.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/Robot.java]: This is for logging to a USB, so unless we would plan on using the USB logs that come off of driver practice or testing, we shouldn't have uneccessary logging going on

File: .gitignore
Comment (mimizh2418) [reply .gitignore:4]: it was a little annoying

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java
Comment (rutmanz) [reply src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java:23]: The sim controller might be inverted, but that seems odd. Make sure you test this when you get the robot

File: src/main/java/org/team1540/bunnybotTank2023/io/drivetrain/DrivetrainIOSim.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/io/drivetrain/DrivetrainIOSim.java:33]: Ok in my defense, this is what 6328 did in their actual competition code, also it's just simulation so its probably fine

File: src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java
Comment (mimizh2418) [reply src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java:23]: I just tested the controller on the actual DS app, the joysticks axes are in fact inverted, although this might be a skill issue with my controller since it's an xbox 1 controller vs an xbox series controller

Comment (rutmanz) [reply src/main/java/org/team1540/bunnybotTank2023/commands/drivetrain/TankdriveCommand.java:23]: So it sounds like your controller is just weird then, you might be able to invert in the simulator gui, but you should flip this back before merging.

**Review by rutmanz — APPROVED** [review]

**Pull request merged**
